apps/token/src/routes/governance/components/list-asset/list-asset.tsx

```diff
@@ -0,0 +1,111 @@
+import type { CollateralBridge } from '@vegaprotocol/smart-contracts';
+import { AssetStatus } from '@vegaprotocol/types';
+import { Button } from '@vegaprotocol/ui-toolkit';
+import { useBridgeContract, useEthereumTransaction } from '@vegaprotocol/web3';
+import { useTranslation } from 'react-i18next';
+import {
+  useAssetListBundleQuery,
+  useProposalAssetQuery,
+} from './__generated___/Asset';
+import { EthWalletContainer } from '../../../../components/eth-wallet-container';
+
+const useListAsset = (assetId: string) => {
+  const bridgeContract = useBridgeContract();
+
+  const transaction = useEthereumTransaction<CollateralBridge, 'list_asset'>(
+    bridgeContract,
+    'list_asset'
+  );
+  const {
+    data,
+    loading: loadingAsset,
+    error: errorAsset,
+  } = useProposalAssetQuery({
+    variables: {
+      assetId,
+    },
+  });
+
+  const {
+    data: assetData,
+    loading: loadingBundle,
+    error: errorBundle,
+  } = useAssetListBundleQuery({
+    variables: {
+      assetId,
+    },
+    skip: !data,
+  });
+  return {
+    transaction,
+    data,
+    loadingAsset,
+    errorAsset,
+    assetData,
+    loadingBundle,
+    errorBundle,
+  };
+};
+
+export interface ListAssetProps {
+  lifetimeLimit: string;
+  withdrawalThreshold: string;
+  assetId: string;
+}
+
+export const ListAsset = ({
+  assetId,
+  lifetimeLimit,
+  withdrawalThreshold,
+}: ListAssetProps) => {
+  const { t } = useTranslation();
+  const {
+    transaction,
+    data,
+    loadingAsset,
+    errorAsset,
+    assetData,
+    loadingBundle,
+    errorBundle,
+  } = useListAsset(assetId);
+  const { perform, Dialog } = transaction;
+
+  if (
+    !assetData?.erc20ListAssetBundle ||
+    !assetData.erc20ListAssetBundle ||
+    !data?.asset ||
+    loadingAsset ||
+    loadingBundle
+  ) {
+    return null;
+  }
+  if (data.asset.source.__typename !== 'ERC20') return null;
+  if (data.asset.status !== AssetStatus.STATUS_PENDING_LISTING) return null;
+  if (errorAsset || errorBundle) return null;
```

Comment:
Chore (non-blocking) looks like at least some of this custom logic can be simplified by using the AsyncRenderer component

Reply:
I deliberately am avoiding async renderer here because it's a minute use case (like 99% of proposal will not render this component) so I didn't want a loading spinner that cause the content to jump around or delay the loading of the rest of the page until this was done. So I thought it best that this just adds itself, if the component is in that state. Happy to move it different behaviour if you think something would be better though?

Reply:
Ooh, that's a good point. That makes sense to me!